### PR TITLE
[WIP] Fix ris import

### PR DESCRIPTION
**Pull request by tschechlovdev**
Regarding: #1647 
Doi import should now work with the DO field. Also I've added some fields, that are available in the [RIS specification](http://referencemanager.com/sites/rm/files/m/direct_export_ris.pdf). I've also done some code cleanups.
#1074 should be also fixed now. I've added the scopus file as test file.

- [x] Tests created for changes
- [x] Manually tested changed features in running JabRef

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/importer/fileformat/RisImporter.java [hunk 1]
```diff
@@ -43,28 +45,17 @@ public String getDescription() {
...
 
-        String line;
-        while ((line = reader.readLine()) != null) {
-            sb.append(line);
-            sb.append('\n');
-        }
+        String linesAsString = reader.lines().reduce((line, nextline) -> line + "\n" + nextline).get();
 
-        String[] entries = sb.toString().replace("\u2013", "-").replace("\u2014", "--").replace("\u2015", "--")
+        String[] entries = linesAsString.replace("\u2013", "-").replace("\u2014", "--").replace("\u2015", "--")
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/importer/fileformat/RisImporter.java]: Maybe you could extract that method to the StringUtil class, e.g. call it ReplaceUnicodeDashes..

File: src/test/resources/net/sf/jabref/logic/importer/fileformat/RisImporterTestDoiAndJournalTitle.bib [hunk 2]
```diff
@@ -0,0 +1,10 @@
+@patent{,
+  author   = {some author},
+  month    = dec,
+  year     = {2014},
+  caption  = {254},
+  edition  = {5},
+  editor   = {some editor},
+  language = {eng},
+  number   = {25},
+}
```
Comment (boceckts) [inline src/test/resources/net/sf/jabref/logic/importer/fileformat/RisImporterTestDoiAndJournalTitle.bib]: The name of the test files confuses me a lot...Isn't the purpose here to test the journal title and doi fields? So why are the fields not present in the file?

File: src/test/resources/net/sf/jabref/logic/importer/fileformat/RisImporterTest3.bib [hunk 3]
```diff
@@ -1,14 +1,14 @@
 @incollection{,
   author = {Barata, Catarina and Celebi, MEmre and Marques, JorgeS},
-  booktitle = {Dermoscopy Image Analysis},
   comment = {doi:10.1201/b19107-2},
   doi = {10.1201/b19107-2},
```
Comment (boceckts) [inline src/test/resources/net/sf/jabref/logic/importer/fileformat/RisImporterTest3.bib]: I have seen that RisImporterTest3.ris specifies the both doi fields (M3 and DO) with the same doi string. Please move one of them to a different or new test file. This way the tests would have already showed that the DO field didn't import the doi.

File: src/test/resources/net/sf/jabref/logic/importer/fileformat/RisImporterTestDoiAndJournalTitle.bib (hunk 2 above)
Comment (tschechlovdev) [inline src/test/resources/net/sf/jabref/logic/importer/fileformat/RisImporterTestDoiAndJournalTitle.bib]: Yeah you're right... Don't know how I could forget them :smile:

File: src/test/resources/net/sf/jabref/logic/importer/fileformat/RisImporterTest3.bib (hunk 3 above)
Comment (tschechlovdev) [inline src/test/resources/net/sf/jabref/logic/importer/fileformat/RisImporterTest3.bib]: I've used the DO field in the TestDoiAndJournalFile.

**Comment by tschechlovdev** [comment]
Travis fails, because the Architecture tests say, that the StringUtil class should be kept as small as possible ...

**Comment by Siedlerchr** [comment]
Speak with @lenhard if it's okay to increase it and how to fix the test

**Comment by tschechlovdev** [comment]
I tried to contact @lenhard, but didn't get an answer. So for the moment I think it is the best to remove the method from the StringUtil class, so that this PR don't have to wait to long. In addition it would only be 2 classes who would use this method.

Travis is still failing because of a DOI test.

**Comment by Siedlerchr** [comment]
Yes, it is the proper way for the moment. We can think later on for moving it.
I would merge it in then when you have done it

**Comment by lenhard** [comment]
@tschechlovdev You contacted me? How? :-) This mention of myself is the first sign I see, but maybe I overlooked something.

**Comment by Siedlerchr** [comment]
@lenhard  The question was about the "Failing travis test" regarding the StringUtil method.
The ideas was to add  a method for replacing unicode dashes:
` String[] entries = linesAsString.replace("\u2013", "-").replace("\u2014", "--").replace("\u2015", "--") `
which is also used in medline Plain.

**Comment by tobiasdiez** [comment]
I know I'm late to the party, but one question: is this unicode dash needed for the page numbers? Because then there is a cleanup formatter which does the job (Page number normalizer, or some permutation of these words).

**Comment by lenhard** [comment]
The reason for moving more functionality to `StringUtil` in model was, if I recall it correctly, that the group fixes needed these methods. If the dash replacement functionality is not needed by model classes, why should it go into that package? If you need that in multiple importer classes, why not introduce a util class for the importer package?

**Pull request merged**